### PR TITLE
feat(spark-jobs): Define a basic Spark job which copies chunks between Cassandra stores.

### PR DESCRIPTION
**Pull request by broneill**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 
New feature.

### CODE REVIEW

File: spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  val targetConfig = openConfig("spark.filodb.chunkcopier.target.configFile")
+
+  val sourceCassConfig = sourceConfig.getConfig("cassandra")
+  val targetCassConfig = targetConfig.getConfig("cassandra")
+
+  val sourceDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.chunkcopier.source.dataset"))
+  val targetDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.chunkcopier.target.dataset"))
+
+  // Examples: 2019-10-20T12:34:56Z  or  2019-10-20T12:34:56-08:00
+  private def parseDateTime(str: String) = Instant.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(str))
```
Comment (vishramachandran) [inline spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: Do you mind making this configuration consistent with DownsamplerMain? Perhaps the following line can also use the above approach ?

```
    val userTimeInPeriod: Long = spark.sparkContext.getConf.get("spark.filodb.downsampler.userTimeOverride",
      s"${System.currentTimeMillis() - settings.downsampleChunkDuration}").toLong
```

File: spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+  val ingestionTimeStart = parseDateTime(conf.get("spark.filodb.chunkcopier.ingestionTimeStart"))
+  val ingestionTimeEnd = parseDateTime(conf.get("spark.filodb.chunkcopier.ingestionTimeEnd"))
+
+  val diskTimeToLiveSeconds = conf.getTimeAsSeconds("spark.filodb.chunkcopier.diskTimeToLive")
+  val splitsPerNode = conf.getInt("spark.filodb.chunkcopier.splitsPerNode", 1)
+  val batchSize = conf.getInt("spark.filodb.chunkcopier.batchSize", 10000)
+
+  val readSched = Scheduler.io("cass-read-sched")
+  val writeSched = Scheduler.io("cass-write-sched")
```
Comment (vishramachandran) [inline spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: Reuse from DownsampleContext? Or rename that class as SparkJobContext ?

File: spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      .getOrCreate()
+
+    val splits = copier.getScanSplits
+
+    logger.info(s"Cassandra split size: ${splits.size}. We will have this many spark partitions. " +
+      s"Tune splitsPerNode which was $copier.splitsPerNode if parallelism is low")
+
+    spark.sparkContext
+      .makeRDD(splits)
+      .foreachPartition(splitIter => ChunkCopier.lookup(conf).run(splitIter))
```
Comment (vishramachandran) [inline spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: Did you verify that `conf` is same for each worker task ? For example, no additional properties are added/removed/updated ?

Comment (broneill) [reply spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: Didn't check. Do you expect any to be added?

File: spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala (hunk 1 above)
Comment (broneill) [reply spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: Which configuration consistent? You'd prefer that the timestamp be specified as seconds from 1970? The userTimeInPeriod line is capturing a duration, not an instant, and so it's different.

File: spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala (hunk 3 above)
Comment (broneill) [reply spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: The SparkConf docs state that the config cannot change after running.

File: spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala (hunk 2 above)
Comment (broneill) [reply spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: The design is a bit different here. I wanted to have the entire state encapsulated into one thing, and it's cached by a different key. I'm not sure which approach is better. I like the fact that the chunk copier code doesn't have to have transient fields all over the place, making it a bit simpler to understand and (hopefully) maintain.

File: spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala (hunk 1 above)
Comment (vishramachandran) [reply spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: I would like userTimeInPeriod to be specified as a string like in your PR

Comment (broneill) [reply spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: But that's a duration, and it has a different syntax. Duration syntax is less standardized and many are unfamiliar with it.

Comment (broneill) [reply spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala]: Will fix in a separate PR.

**Pull request merged**